SpatialGDK/Source/Private/Utils/EntityRegistry.cpp

```diff
@@ -27,11 +27,32 @@ void UEntityRegistry::RemoveFromRegistry(const Worker_EntityId& EntityId)
 
 void UEntityRegistry::RemoveFromRegistryImpl(const AActor* Actor, const Worker_EntityId& EntityId)
 {
-	ActorToEntityId.Remove(Actor);
+	if (Actor)
+	{
+		if (ActorToEntityId.Contains(Actor))
+		{
+			ActorToEntityId.Remove(Actor);
+		}
+		else
+		{
+			UE_LOG(LogEntityRegistry, Warning, TEXT("Tried to remove actor %s from registry but it wasn't there"), *Actor->GetFullName());
+		}
+	}
+	else
+	{
+		UE_LOG(LogEntityRegistry, Warning, TEXT("Couldn't remove Actor from registry: Actor == nullptr"));
+	}
 
 	if (EntityId != -1)
```

Comment:
Do we have a constant for invalid entity id?

Reply:
Yes, but it doesn't actually cover all invalid entity IDs (must be > 0). Changed to check for that instead